feat: add version and version picker

tinayuangao opened the pull request:
- add version picker (using menu)
- make stackblitz use the same version as the doc site


Demo: https://material2-docs-dev.firebaseapp.com/

mmalerba commented on src/app/shared/version/version.ts:
```diff
@@ -0,0 +1,23 @@
+    title: '5.2.4'
+  },
+  {
+    url: `http://material2-docs-dev.firebaseapp.com`,
```
is this supposed to be pointed at the dev one?

tinayuangao replied:
No... I will update the url once we set up the domain for v5.

mmalerba commented on src/app/shared/version-picker/version-picker.ts:
```diff
@@ -0,0 +1,29 @@
+
+  materialVersion = materialVersion;
+
+  displayVersion = materialVersion.match(/\d+.\d+.\d+/)[0];
```
extract the regex into a var

tinayuangao replied:
Done. Thanks for review!

tinayuangao commented:
Updated the pr with actual urls. 
Added npm scripts for v5-dev and v5-prod.

The v5 is available https://v5.material.angular.io/ (without version picker)

jelbourn commented on package.json:
```diff
@@ -15,7 +15,9 @@
-    "publish-dev": "npm run build-themes && ng build --aot --prod && firebase use material2-docs-dev && firebase deploy"
+    "publish-dev": "npm run build-themes && ng build --aot --prod && firebase use material2-docs-dev && firebase deploy",
+    "publish-v5-prod": "npm run build-themes && ng build --aot --prod && firebase use v5-material-angular-io && firebase deploy",
+    "publish-v5-dev": "npm run build-themes && ng build --aot --prod && firebase use material2-docs-5 && firebase deploy"
```
File an issue saying we should create a deploy script for this rather than adding more npm commands over time?

tinayuangao replied:
Added file `tools/deploy.sh` and updated `package.json`
